dipy/reconst/ivim.py

```diff
@@ -0,0 +1,391 @@
+
+        split_b : float
+            The b-value to split the data on for two-stage fit
+
+        bounds : array, (4,4)
+            Bounds for the parameters. This is applicable only for Scipy
+            version > 0.17 as we use least_squares for the fitting which
+            supports bounds. For versions less than Scipy 0.17, this is
+            by default set to None. Setting a bound on a Scipy version less
+            than 0.17 will raise an error.
+
+            This can be supplied as a tuple for Scipy versions 0.17. It is
+            recommended to upgrade to Scipy 0.17 for bounded fitting. The
+            default bounds are set to ([0., 0., 0., 0.], [np.inf, 1., 1., 1.])
+
+        tol : float, optional
+            Default : 1e-7
+            Tolerance for convergence of minimization.
+
+        options : dict, optional
+            Dictionary containing gtol, ftol, eps and maxiter. This is passed
+            to leastsq. By default these values are set to
+            options={'gtol': 1e-7, 'ftol': 1e-7, 'eps': 1e-7, 'maxiter': 1000}
+
+        References
+        ----------
+        .. [1] Le Bihan, Denis, et al. "Separation of diffusion
+                   and perfusion in intravoxel incoherent motion MR
+                   imaging." Radiology 168.2 (1988): 497-505.
+        .. [2] Federau, Christian, et al. "Quantitative measurement
+                   of brain perfusion with intravoxel incoherent motion
+                   MR imaging." Radiology 265.3 (2012): 874-881.
+        """
+        ReconstModel.__init__(self, gtab)
+        self.split_b = split_b
+        self.bounds = bounds
+        self.tol = tol
+        self.options = options
+
+        if SCIPY_LESS_0_17 and self.bounds is not None:
+            e_s = "Scipy versions less than 0.17 do not support "
+            e_s += "bounds. Please update to Scipy 0.17 to use bounds"
+            raise ValueError(e_s)
+        else:
+            self.bounds = (np.array([0., 0., 0., 0.]),
+                           np.array([np.inf, 1., 0.1, 0.1]))
+
+    @multi_voxel_fit
+    def fit(self, data, mask=None):
+        """ Fit method of the Ivim model class
+
+        Parameters
+        ----------
+        data : array
+            The measured signal from one voxel. A multi voxel decorator
+            will be applied to this fit method to scale it and apply it
+            to multiple voxels.
+
+        mask : array
+            A boolean array used to mark the coordinates in the data that
+            should be analyzed that has the shape data.shape[:-1]
+
+        Returns
+        -------
+        IvimFit object
+
+        References
+        ----------
+        .. [1] Federau, Christian, et al. "Quantitative measurement
+                   of brain perfusion with intravoxel incoherent motion
+                   MR imaging." Radiology 265.3 (2012): 874-881.
+        """
+        # Call the function _estimate_S0_D to get initial x0 guess.
```

Comment:
Looks like you call estimate_x0, not _estimate_S0_D, so the comment is incorrect.